// Copyright 2017 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package ray.rpc;

import "src/ray/protobuf/common.proto";

enum JobStatus {
  // Job OK.
  OK = 0;
  // Job failed.
  FAILED = 1;
}

message DispatchJobInfoRequest {
  // The job ID.
  string job_id = 1;
  // Start driver or not.
  bool start_driver = 2;
}

message DispatchJobInfoReply {
  // The job ID.
  string job_id = 1;
  // Current status of dispatched job.
  JobStatus status = 2;
}

// Service for communicating with the job_agent.py process on a remote node.
service JobService {
  // Dispatch job info.
  rpc DispatchJobInfo(DispatchJobInfoRequest) returns (DispatchJobInfoReply);
}
